uni-stark/src/symbolic_variable.rs

```diff
@@ -1,26 +1,56 @@
 use core::marker::PhantomData;
 use core::ops::{Add, Mul, Sub};
 
-use p3_field::Field;
+use p3_field::{ExtensionField, Field};
 
 use crate::symbolic_expression::SymbolicExpression;
 
+#[derive(Debug, Clone, Copy)]
+pub enum Entry {
```

Comment:
Not sure about grouping these together, it would make some invalid things (like challenges with `is_next=1`) representable in the type system.

We could make `Public` and `Challenge` separate cases in `SymbolicExpression`, or if you think `SymbolicExpression` is getting too large, maybe rename `SymbolicVariable` -> `TraceVariable` and introduce a broader `SymbolicVariable` as a sub-enum of `SymbolicExpression`?

Reply:
Yes, that's not ideal. I was thinking of doing 

```rust
pub enum Entry {
   Preprocessed(is_next: bool)
   Main(is_next : bool),
   Perm(is_next: bool)
   Challenge,
   Public,
}
```

Reply:
Yeah that seems better. I think it'd also be a bit funny to use `column` with `Challenge`/`Public` since they're not really matrices, but I suppose `index` would be an okay name for everything.